server/src/test/java/org/elasticsearch/index/mapper/DateFieldMapperTests.java

```diff
@@ -583,12 +583,12 @@ protected SyntheticSourceSupport syntheticSourceSupport() {
                 : DateFieldMapper.DEFAULT_DATE_TIME_NANOS_FORMATTER;
 
             @Override
-            public SyntheticSourceExample example() {
+            public SyntheticSourceExample example(int maxValues) {
```

Comment:
Is this directly related to the main change or is it part of a separate refactor?

Reply:
Related. I can cut it to it's own change. But it's related.

The new behavior only kicks in if you fetch more than one doc. And, in general, now that there is more state on the SourceLoader it's important to test more than one doc. So I wrote that "more than one doc" version of the test. To test the new behavior and out of paranoia. And *that* needs this. It needs it so that the "more than one doc" can consistently force singleton values. If we didn't have this then it'd be *super* rare to get singletons. But with this the "more than one doc" test can do `int maxValue = randomBoolean() ? 1 : 5;` up front. And gets singletons half the time.

Reply:
+1, sounds sensible to me, I'm happy to keep it as part of the same change